api/src/main/java/io/opentelemetry/trace/SpanContext.java

```diff
@@ -75,25 +76,88 @@ public static SpanContext create(
    * @since 0.1.0
    */
   public static SpanContext createFromRemoteParent(
-      TraceId traceId, SpanId spanId, TraceFlags traceFlags, TraceState traceState) {
-    return new AutoValue_SpanContext(traceId, spanId, traceFlags, traceState, /* remote=*/ true);
+      CharSequence traceId, CharSequence spanId, TraceFlags traceFlags, TraceState traceState) {
+    return create(traceId, 0, spanId, 0, traceFlags, traceState, /* remote=*/ true);
   }
 
+  /**
+   * Creates a new {@code SpanContext} with the given identifiers and options.
+   *
+   * @param traceId a CharSequence containing the trace identifier of the span context.
+   * @param traceIdOffset the offset at which the traceId starts.
+   * @param spanId the CharSequence containing the the span identifier of the span context.
+   * @param spanIdOffset the offset at which the spanId starts.
+   * @param traceFlags the trace options for the span context.
+   * @param traceState the trace state for the span context.
+   * @return a new {@code SpanContext} with the given identifiers and options.
+   * @since 0.1.0
+   */
+  @SuppressWarnings({
+    "InconsistentOverloads",
+    "UngroupedOverloads",
+    "OverloadMethodsDeclarationOrder"
+  })
+  public static SpanContext create(
+      CharSequence traceId,
```

Comment:
Since our accessors (e.g. `getTraceId()` use `String`, I don't think there's an advantage to using `CharSequence` in the factories. Can we use `String` everywhere?

Reply:
My thinking (and it wasn't very strong thinking) was that to be more accepting of inputs, and more restrictive on outputs. But, I definitely don't feel too strongly about that.

Reply:
Got it - I'm ok with either, but have a small feeling that `String` feels more familiar to developers

Reply:
switched all to String now. :)